index.js

```diff
@@ -161,9 +177,16 @@ tslintPlugin.report = function(reporter, options) {
     var throwErrors = function() {
         // Throw error
         if (options && options.emitError === true && errorFiles.length > 0) {
-            return this.emit('error', new PluginError('gulp-tslint', 'Failed to lint: ' + allFailures.map(function(failure) {
+            var start = options.reportLimit >= 0 ? options.reportLimit : 0;
```

Comment:
It's cleaner to set start to 0, and then using 'if' to set to a special value. Also rename the variable, perhaps something like 'failuresStartIndex' or something.